reactor-core/src/test/java/reactor/core/publisher/FluxReplayTest.java

```diff
@@ -512,4 +514,32 @@ public void scanSubscriber() {
         test.cancelled = true;
         Assertions.assertThat(test.scan(Scannable.Attr.CANCELLED)).isTrue();
     }
+
+	@Test
+	public void cacheSingleSubscriberWithMultipleRequestsDoesntHang() {
+		Awaitility.await()
+		          .atMost(5, TimeUnit.SECONDS)
+		          .untilAsserted(() -> assertThat(Flux.range(0, 1000)
```

Comment:
nit: `() -> {}` could help reducing the indentation dramatically :)

Reply:
improved the indentation even more dramatically